core/src/ru/spbau/blackout/abilities/gravity/GravityObject.java

```diff
@@ -0,0 +1,86 @@
+package ru.spbau.blackout.abilities.gravity;
+
+import com.badlogic.gdx.math.Vector2;
+
+import ru.spbau.blackout.abilities.simpleshell.SimpleShellObject;
+import ru.spbau.blackout.entities.Damageable;
+import ru.spbau.blackout.entities.DynamicObject;
+import ru.spbau.blackout.entities.GameObject;
+import ru.spbau.blackout.utils.Uid;
+import ru.spbau.blackout.worlds.ServerGameWorld;
+
+import static ru.spbau.blackout.abilities.gravity.GravityAbility.CAST_SOUND_PATH;
+import static ru.spbau.blackout.abilities.gravity.GravityAbility.EFFECT_PATH;
+import static ru.spbau.blackout.abilities.gravity.GravityAbility.RADIUS;
+import static ru.spbau.blackout.abilities.gravity.GravityAbility.SHELL_MASS;
+import static ru.spbau.blackout.abilities.gravity.GravityAbility.TIME_TO_LIVE;
+import static ru.spbau.blackout.utils.Utils.isZeroVec;
+
+public final class GravityObject extends SimpleShellObject {
+
+    private final GameObject caster;
+    private final float maxForce;
+    private final float maxDamage;
+
+    protected GravityObject(Definition def, Uid uid, float x, float y) {
+        super(def, uid, x, y);
+        caster = def.caster;
+        maxForce = def.maxForce;
+        maxDamage = def.damage;
+    }
+
+    @Override
+    public void updateState(float delta) {
+        super.updateState(delta);
+        for (GameObject go : getDef().getContext().gameWorld().getGameObjects()) {
+            if (go != caster && !go.isDead() && go instanceof Damageable) {
+                ((Damageable) go).damage(maxDamage * powerFactor(go.getPosition().dst(getPosition())));
+            }
+        }
+    }
+
+    @Override
+    public void updateBeforeSecondStep() {
+        super.updateBeforeSecondStep();
+        for (GameObject go : getDef().getContext().gameWorld().getGameObjects()) {
+            if (go != caster && go instanceof DynamicObject) {
+                Vector2 offset = getPosition().cpy().sub(go.getPosition());
+                if (!isZeroVec(offset)) {
```

Comment:
This whole part is meant only for dynamic objects. So, it's better to check it in the beginning.